File: app/src/main/java/com/happs/ximand/ringcontrol/viewmodel/dto/InputAlertDialogDto.java
package com.happs.ximand.ringcontrol.viewmodel.dto;

import com.happs.ximand.ringcontrol.OnEventListener;

public class InputAlertDialogDto {

    private int titleResId;
    private OnEventListener<String> inputCompleteListener;
    private OnEventListener<Void> cancelListener;

    public int getTitleResId() {
        return titleResId;
    }

    public InputAlertDialogDto setTitleResId(int titleResId) {
        this.titleResId = titleResId;
        return this;
    }

    public OnEventListener<String> getInputCompleteListener() {
        return inputCompleteListener;
    }

    public InputAlertDialogDto setInputCompleteListener(OnEventListener<String> inputCompleteListener) {
        this.inputCompleteListener = inputCompleteListener;
        return this;
    }

    public OnEventListener<Void> getCancelListener() {
        return cancelListener;
    }

    public InputAlertDialogDto setCancelListener(OnEventListener<Void> cancelListener) {
        this.cancelListener = cancelListener;
        return this;
    }
}

File: app/src/main/java/com/happs/ximand/ringcontrol/viewmodel/dto/InputAlertDialogDto.kt
package com.happs.ximand.ringcontrol.viewmodel.dto

class InputAlertDialogDto {

    var titleResId = 0
        private set
    var inputCompleteListener: ((input: String) -> Unit)? = null
        private set
    var cancelListener: (() -> Unit)? = null
        private set

    fun setTitleResId(titleResId: Int): InputAlertDialogDto {
        this.titleResId = titleResId
        return this
    }

    fun setInputCompleteListener(onInput: ((input: String) -> Unit)?): InputAlertDialogDto {
        this.inputCompleteListener = onInput
        return this
    }

    fun setCancelListener(onCancel: (() -> Unit)?): InputAlertDialogDto {
        this.cancelListener = onCancel
        return this
    }
}